src/main/rx/cc2500_frsky_x.c

```diff
@@ -281,18 +281,13 @@ void frSkyXSetRcData(uint16_t *rcData, const uint8_t *packet)
     c[6] = (uint16_t)((packet[19] << 8) & 0xF00) | packet[18];
     c[7] = (uint16_t)((packet[20] << 4) & 0xFF0) | (packet[19] >> 4);
 
-    uint8_t j = 0;
-    for(uint8_t i = 0; i < 8; i++) {
-        if(c[i] > 2047) {
-            j = 8;
-            c[i] = c[i] - 2048;
-        } else {
-            j = 0;
-        }
-        int16_t temp = (((c[i] - 64) << 1) / 3 + 860);
-        if ((temp > 800) && (temp < 2200)) {
-            rcData[i+j] = temp;
+    for (unsigned i = 0; i < 8; i++) {
+        bool channelIsShifted = false;
+        if (c[i] & 0x800) {
+            channelIsShifted = true;
         }
+        const uint16_t channelValue = c[i] & 0x7FF;
+        rcData[channelIsShifted ? i + 8 : i] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
```

Comment:
int/float conversion is a bit expensive, this should work in integer math:
`(channelValue * 2 - 64 * 2 + 860 * 3 ) / 3;`